crates/oxc_transformer/src/plugins/styled_components.rs

```diff
@@ -0,0 +1,1154 @@
+    /// tree-shaking in some bundlers.
+    fn default() -> Self {
+        Self {
+            display_name: true,
+            file_name: true,
+            ssr: true,
+            transpile_template_literals: true,
+            pure: false,
+            minify: true,
+            namespace: None,
+            css_prop: true,
+            meaningless_file_names: default_for_meaningless_file_names(),
+            top_level_import_paths: vec![],
+        }
+    }
+}
+
+/// Tracks symbol IDs for styled-components imports to identify which variables
+/// are bound to styled-components functionality.
+#[derive(Default)]
+struct StyledComponentsBinding {
+    /// `import * as styled from 'styled-components'`
+    namespace: Option<SymbolId>,
+    /// `import styled from 'styled-components'` or `import { default as styled } from 'styled-components'`
+    styled: Option<SymbolId>,
+    /// Named imports like `import { createGlobalStyle, css, keyframes } from 'styled-components'`
+    helpers: FxHashMap<String, SymbolId>,
+}
+
+pub struct StyledComponents<'a, 'ctx> {
+    pub options: StyledComponentsOptions,
+    pub ctx: &'ctx TransformCtx<'a>,
+
+    // State
+    /// Tracks which variables are bound to styled-components imports
+    styled_bindings: StyledComponentsBinding,
+    /// Counter for generating unique component IDs
+    component_count: usize,
+    /// Hash of the current file for component ID generation
+    file_hash: Option<&'a str>,
+}
+
+impl<'a, 'ctx> StyledComponents<'a, 'ctx> {
+    pub fn new(options: StyledComponentsOptions, ctx: &'ctx TransformCtx<'a>) -> Self {
+        Self {
+            options,
+            ctx,
+            styled_bindings: StyledComponentsBinding::default(),
+            file_hash: None,
+            component_count: 0,
+        }
+    }
+}
+
+impl<'a> Traverse<'a, TransformState<'a>> for StyledComponents<'a, '_> {
+    fn enter_program(&mut self, program: &mut Program<'a>, ctx: &mut TraverseCtx<'a>) {
+        self.collect_styled_bindings(program, ctx);
+    }
+
+    fn enter_variable_declarator(
+        &mut self,
+        variable_declarator: &mut VariableDeclarator<'a>,
+        ctx: &mut TraverseCtx<'a>,
+    ) {
+        self.handle_pure_annotation(variable_declarator, ctx);
+    }
+
+    fn enter_expression(&mut self, expr: &mut Expression<'a>, ctx: &mut TraverseCtx<'a>) {
+        if matches!(expr, Expression::TaggedTemplateExpression(_)) {
+            self.transform_tagged_template_expression(expr, ctx);
+        }
+    }
+
+    fn enter_call_expression(&mut self, call: &mut CallExpression<'a>, ctx: &mut TraverseCtx<'a>) {
+        if (self.options.display_name || self.options.ssr)
+            // Only transform call expression that is not a part of a member expression
+            // or a callee of another call expression.
+            && !matches!(
+                ctx.parent(),
+                Ancestor::CallExpressionCallee(_) | Ancestor::StaticMemberExpressionObject(_)
+            )
```

Comment:
Should `ComputedMemberExpression` be transformed?

Reply:
Aha, it should be, I miss to support `ComputedMemberExpression`. I will do it in the follow-up